File: rzops-api/domain/src/models/monitor_target.rs
use chrono::{DateTime, Utc};
use uuid::Uuid;


/// MonitorTarget entity — maps to cmdb_monitor_target.
#[derive(Debug, Clone)]
pub struct MonitorTarget {
    pub id: Uuid,
    pub name: String,
    pub site_id: Option<Uuid>,
    pub target_type: Option<String>,
    pub target_id: Option<Uuid>,
    pub monitor_type: Option<String>,
    pub endpoint: Option<String>,
    pub interval_seconds: Option<i32>,
    pub status: String,
    pub remarks: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

File: rzops-web/src/lib/types/monitor_target.ts
export interface MonitorTargetResponse {
  id: string;
  name: string;
  target_type: string | null;
  target_id: string | null;
  target_name: string | null;
  monitor_type: string | null;
  endpoint: string | null;
  interval_seconds: number | null;
  status: string;
  remarks: string | null;
  created_at: string;
  updated_at: string;
}

export interface MonitorTargetListResponse {
  data: MonitorTargetResponse[];
  count: number;
}

export interface CreateMonitorTargetRequest {
  name: string;
  target_type?: string;
  target_id?: string;
  monitor_type?: string;
  endpoint?: string;
  interval_seconds?: number;
  status?: string;
  remarks?: string;
}

export interface UpdateMonitorTargetRequest extends Partial<CreateMonitorTargetRequest> {}

export interface ListMonitorTargetsQuery {
  q?: string;
  status?: string;
  target_type?: string;
  target_id?: string;
  page?: number;
  per_page?: number;
}
